### ISSUE TITLE
Package uhd fails to build

### ISSUE THREAD

**Issue by markuskowa** [NixOS/nixpkgs#33549]
## Issue description
Building package uhd against master (d1c81e29c35365f17f165af4783a7af77ee50d46) fails.
Error message:
```
/tmp/nix-build-uhd-3.10.2.0.drv-0/source/host/lib/transport/udp_zero_copy.cpp: In constructor 'udp_zero_copy_asio_impl::udp_zero_copy_asio_impl(const string&, const string&, const uhd::transport::zero_copy_xport_params&)':
/tmp/nix-build-uhd-3.10.2.0.drv-0/source/host/lib/transport/udp_zero_copy.cpp:193:29: error: 'class boost::asio::basic_datagram_socket<boost::asio::ip::udp>' has no member named 'native'
         _sock_fd = _socket->native();
```

The latest hydra build (https://hydra.nixos.org/build/67133025) also fails with the same message.

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.71, NixOS, 17.09.2450.3d1a9485e28 (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 1.11.16`

**Comment by lukeadams** [comment]
For tracking purposes: [Darwin fails as well](https://hydra.nixos.org/build/67130277)

**Comment by markuskowa** [comment]
Fix by 9c6a17c5243d785c345a681ada4f0a4e9227953d.
